### ISSUE TITLE
Update dependencies as minimatch@2.0.10 is vulnerable to Regular Expression Denial of Service (aka ReDOS)

### ISSUE THREAD

**Issue by michaelhidalgo** [elastic/generator-kibana-plugin#43]
As stated in this advisory (https://nodesecurity.io/advisories/118) minimatch <=3.0.1 is vulnerable to RegexDOS, and it seems like that version is one of the dependencies generator-kibana relies at the moment.

You can confirm that by running the following command inside the project containing the plugin

```
Michaels-MacBook-Pro:plugin michaelhidalgo$ ls
README.md	node_modules	public
index.js	package.json	server

npm ls minimatch

```

``` Michaels-MacBook-Pro:plugin michaelhidalgo$ npm ls minimatch
demo@0.0.0 /Users/michaelhidalgo/Desktop/elastic/plugin
├─┬ babel-eslint@4.1.8
│ └─┬ babel-core@5.8.38
│   └── minimatch@2.0.10 
└─┬ eslint@1.10.3
  └── minimatch@3.0.3

```
So one of the dependencies of babel depends on the version that might be vulnerable.

**Comment by w33ble** [comment]
Thanks for letting us know. The problem is our use of an outdated version of babel, which is a larger issue unfortunately.

I think that because that vulnerable version of minimatch is limited to babel, the code at runtime is still safe. My guess is that babel uses minimatch for its process, and should not affect the code it outputs, but I haven't dug into it, so I could be wrong.

**Comment by michaelhidalgo** [comment]
Yes you are right, if you drill down the search using npm, an outdated version of babel is the one who has it as a dependency.

```
Michaels-MacBook-Pro:plugin michaelhidalgo$ npm ls minimatch@2.0.10
sekurity@0.0.0 /Users/michaelhidalgo/Desktop/elastic/plugin
└─┬ babel-eslint@4.1.8
  └─┬ babel-core@5.8.38
    └── minimatch@2.0.10 

```
I believe the real risk is if someone can trigger the Denial of Service condition from up the dependency chain, that is by using babel, maybe is worth to do a PoC and determine if it is exploitable.

**Comment by michaelhidalgo** [comment]
Btw, I found out that the same issue is happening in kibana (master branch) but also the problem is with tough-cookie@2.2.2 (here is the advisory https://nodesecurity.io/advisories/130). But I should open an issue on kibana though.

**Comment by w33ble** [comment]
Now that Kibana is using a newer babel, this package can in fact be updated. 

The template is already updated: https://github.com/elastic/template-kibana-plugin/blob/master/template/package.json

**Comment by michaelhidalgo** [comment]
Cool, I will give it a try and keep you posted.
